### PR TITLE
Some minor clean-ups

### PR DESCRIPTION
**Pull request by punchagan**
Attempting to move towards a clearer library/API. It's a WIP, but sharing what I've done so far.

### CODE REVIEW

File: bin/main.ml [hunk 1]
```diff
@@ -3,49 +3,32 @@ open Prototype
...
-  let revdeps = list_revdeps package no_transitive_revdeps in
-  OpamConsole.msg "Number of reverse dependencies: %d\n"
-    (OpamPackage.Set.cardinal revdeps);
+  let revdeps = Revdeps.list_revdeps pkg no_transitive_revdeps in
 
-  (* Display on console *)
-  OpamPackage.Set.iter
-    (fun pkg -> OpamConsole.msg "%s\n" (OpamPackage.to_string pkg))
-    revdeps;
+  Revdeps.Display.packages revdeps;
```
Comment (shonfeder) [inline bin/main.ml:11]: I wonder if we can move this out of the frontend and just have it in the library? I feel like we can abstract this as a function `pkg -> repo_dir -> options -> revdeps_list`, and the fact that we need to initialize an env as a side effect just seems like an implementation detail (where `options` is are the flags). WDYT?

**Review by shonfeder — COMMENTED** [review]
This really helps me see some of the joints of the code more clearly. It is a helpful cleanup already to me!

I thought it might help if I elaborated a bit what I had in my head when we were talking about carving out APIs to modularize. I am imagining we could end up with a core library with API roughtly like

```ocaml
module type Revdeps_intf = sig
  val list_revdeps : ?local_repo_dir:string -> pkg:string -> transitive_deps:bool -> string list

  type test_err
  val test_revdeps : ?local_repo_dir:string -> pkg:string -> transitive_deps:bool -> test_err option
end
```

And a parameterized implementation like

```ocaml
module Revdeps  (Pkgs : Packages) (Sol : Solver) (Bld : Tester) : Revdeps_intf = struct
  (* ... *)
end
```

Where, e.g, `Packages` would give us stuff like

```ocaml
module type Packages = sig
  (** Package manager interface *)

  type package
  val to_string : package -> string
end
```

And one particular implementation of this would be 

```ocaml
module Opam_packages : Packages = struct
  type package  = OpamPackage.t
  let to_string = OpamPackage.to_string
end
```

etc. This may be where you are headed already, and/or there may also be good reasons why this isn't as feasible as I'd expect. Either way, I thought it may be worth explaining my understanding more explicitly so we can compare our thinking and see where we turn out :)
 
The hope, of course, is that we'd have a core logic into which we could drop in either the Opam package impl or the dune impl, the cudf-based solver impl or some hypothetical future based directly on 0install, etc. I confess I need to look deeper at the current code to see how much logic we may have left after paramaterizing in these ways :D

**Review by shonfeder — APPROVED** [review]

File: bin/main.ml (hunk 1 above)
Comment (punchagan) [reply bin/main.ml:11]: Yes, I was wondering about making the environment setup look more like an implementation detail. Thanks for this suggestion.

**Comment by punchagan** [comment]
These suggestions make sense to me, but I was also wondering that if we really need an abstraction over the OpamPackage type since we are actually dealing with packages in the Opam repository, right now, and Dune also seems to be using vendored Opam and sharing types to represent packages etc. I understand the spirit in which you make the suggestion, but was also vary of doing too much without a clear need for it. I'll try to clean up the interface a little more, based on your suggestions and we can take a look, again. Thanks!

**Comment by shonfeder** [comment]
Good point about the opam package type. Maybe worth looking at dune to see how/if they are abstracting over the opam package logic or usually it directly throughout?

I also agree there is a real risk in premature abstraction! Best to focus just on the api boundaries that are real and pending for us. Thanks for opening the pr, as the review and discussion seemed helpful in clarifying :)

**Pull request merged**